src/theory/quantifiers/sygus/cegis_unif.cpp

```diff
@@ -243,36 +233,85 @@ bool CegisUnif::processConstructCandidates(const std::vector<Node>& enums,
+}
+
+void CegisUnif::setConditions(std::map<Node, std::vector<Node>>& unif_cenums,
+                              std::map<Node, std::vector<Node>>& unif_cvalues,
+                              std::vector<Node>& lems)
+{
+  Node cost_lit = d_u_enum_manager.getAssertedLiteral();
+  NodeManager* nm = NodeManager::currentNM();
   // set the conditions
   for (const Node& c : d_unif_candidates)
   {
     for (const Node& e : d_cand_to_strat_pt[c])
     {
-      d_sygus_unif.setConditions(
-          e, cost_lit, unif_enums[1][e], unif_values[1][e]);
-      // if condition enumerator had value, exclude this value
-      if (options::sygusUnifCondIndependent() && !unif_enums[1][e].empty())
+      Assert(unif_cenums.find(e) != unif_cenums.end());
+      Assert(unif_cvalues.find(e) != unif_cvalues.end());
+      d_sygus_unif.setConditions(e, cost_lit, unif_cenums[e], unif_cvalues[e]);
+      // if condition enumerator had value and it is being passively generated,
+      // exclude this value
+      if (options::sygusUnifCondIndependent() && !unif_cenums[e].empty())
       {
-        Node eu = unif_enums[1][e][0];
+        Node eu = unif_cenums[e][0];
         Assert(d_tds->isEnumerator(eu));
         if (d_tds->isPassiveEnumerator(eu))
         {
           Node g = d_u_enum_manager.getActiveGuardForEnumerator(eu);
-          Node exp_exc =
-              d_tds->getExplain()
-                  ->getExplanationForEquality(eu, unif_values[1][e][0])
-                  .negate();
+          Node exp_exc = d_tds->getExplain()
+                             ->getExplanationForEquality(eu, unif_cvalues[e][0])
+                             .negate();
           lems.push_back(nm->mkNode(OR, g.negate(), exp_exc));
         }
       }
     }
   }
+}
+
+bool CegisUnif::processConstructCandidates(const std::vector<Node>& enums,
+                                           const std::vector<Node>& enum_values,
+                                           const std::vector<Node>& candidates,
+                                           std::vector<Node>& candidate_values,
+                                           bool satisfiedRl,
+                                           std::vector<Node>& lems)
+{
+  if (d_unif_candidates.empty())
+  {
+    Assert(d_non_unif_candidates.size() == candidates.size());
+    return Cegis::processConstructCandidates(
+        enums, enum_values, candidates, candidate_values, satisfiedRl, lems);
+  }
+  // the unification enumerators for conditions and their model values
+  std::map<Node, std::vector<Node>> unif_cenums;
+  std::map<Node, std::vector<Node>> unif_cvalues;
+  if (!satisfiedRl)
+  {
+    // if condition values are being indepedently enumerated, they should be
+    // communicated to the decision tree strategies indepedently of the
+    // evaluation heads valuation satisfying the refinement lemmas
+    if (options::sygusUnifCondIndependent())
+    {
+      setEnumValues(enums, enum_values, unif_cenums, unif_cvalues, lems);
```

Comment:
Can we simplify the control flow a bit here? I'd prefer if there were only one call to setEnumValues/setConditions in this function.